### PR TITLE
LB-323: Allow users to request statistics calculation

### PR DESCRIPTION
**Pull request by paramsingh**
# Summary

* This is a…
    * [ ] Bug fix
    * [x] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other

# Problem

* JIRA ticket (_optional_):  [LB-323](https://tickets.metabrainz.org/browse/LB-323)

# Solution

Add a stats_calculator script and send users to that queue.

### CODE REVIEW

**Comment by paramsingh** [comment]
Renamed the scheduler container because it isn't used in production either. Makes sense, I think.

File: listenbrainz/webserver/views/profile.py [hunk 1]
```diff
@@ -202,3 +212,33 @@ def upload():
...
+            'id': current_user.id,
+            'musicbrainz_id': current_user.musicbrainz_id,
+        }
+        publish_data_to_queue(
+            data=data,
+            exchange=current_app.config['STATS_EXCHANGE'],
+            queue=current_app.config['STATS_QUEUE'],
+            error_msg='Could not put user %s into statistics calculation queue, please try again later',
+        )
+        _redis.redis.set(construct_stats_queue_key(current_user.musicbrainz_id), 'queued')
```
Comment (mayhem) [inline listenbrainz/webserver/views/profile.py]: Now that we merged the brainzutils PR, should this be updated to use brainz util caching instead?

File: listenbrainz/webserver/views/profile.py [hunk 2]
```diff
@@ -202,3 +212,33 @@ def upload():
...
+def request_stats():
+    """ Check if the current user's statistics have been calculated and if not,
+        put them in the stats queue for stats_calculator.
+    """
+    status = _redis.redis.get(construct_stats_queue_key(current_user.musicbrainz_id)) == 'queued'
+    if status == 'queued':
+        flash.info('You have already been added to the stats calculation queue! Please check back later.')
+    elif db_stats.valid_stats_exist(current_user.id):
+        flash.info('Your stats were calculated in the most recent stats calculation interval,'
+            ' please wait until the next interval! We calculate new statistics every Monday at 00:00 UTC.')
```
Comment (mayhem) [inline listenbrainz/webserver/views/profile.py]: This is good, but out of date the second we change the cronjob. I think we should improve this in the future, but roll with it for now.

File: listenbrainz/stats/stats_calculator.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    def start(self):
+        """ Starts the stats calculator. This should run perpetually,
+            monitor the stats rabbitmq queue and calculate stats for
+            entities that it receives from the queue.
+        """
+
+        # if no bigquery support, sleep
+        if not self.app.config['WRITE_TO_BIGQUERY']:
+            while True:
+                time.sleep(10000)
```
Comment (mayhem) [inline listenbrainz/stats/stats_calculator.py]: time has not been imported here.

File: listenbrainz/webserver/views/profile.py (hunk 1 above)
Comment (paramsingh) [reply listenbrainz/webserver/views/profile.py]: This isn't exactly caching which is why I used our RedisConnection directly similar to how we use it for storing now_playing listens.

**Pull request merged**